Description/DescriptionFactory.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2015 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Component\Resource\Description;
+
+use Puli\Repository\Api\Resource\PuliResource;
+
+class DescriptionFactory
+{
+    /**
+     * @var DescriptionEnhancerInterface[]
+     */
+    private $enhancers = [];
+
+    /**
+     * @param array $enhancers
+     */
+    public function __construct(array $enhancers)
+    {
+        $this->enhancers = $enhancers;
+    }
+
+    /**
+     * Return a description of the given (CMF) Resource.
+     *
+     * @param PuliResource $resource
+     */
+    public function getPayloadDescriptionFor(PuliResource $resource)
+    {
+        $description = new Description($resource);
+
+        foreach ($this->enhancers as $enhancer) {
+            if (false === $enhancer->supports($resource)) {
+                continue;
+            }
+
+            $enhancer->enhance($description);
```

Comment:
`$description` is an object, even if you did return it it would still be modified by reference - unless we made it immutable. But I don't really see the benefit.